pkg/stockings/fingopack.go

```diff
@@ -0,0 +1,128 @@
+package stockings
+
+import (
+	"time"
+
+	"github.com/piquette/finance-go"
+	"github.com/piquette/finance-go/chart"
+	"github.com/piquette/finance-go/datetime"
+	"github.com/piquette/finance-go/quote"
+
+	"github.com/shopspring/decimal"
+)
+
+// FingoPack will be used to determine which interface we are using
+type FingoPack struct {
+}
+
+// fingoHistoricalStocks is a list of struct ChartBar which is determined by finance-go
+type piqHistoricalStocks []finance.ChartBar
+
+func (piq FingoPack) GetCurrentPrice(ticker string) (*Stock, error) {
+
+	// simple call to get the quote of a stock in the finance.quote struct
+	quote, err := quote.Get(ticker)
+	if err != nil {
+		return nil, err
+	}
+
+	// need to convert our types to match our universal Stock struct
+	// some of the types are already correct, but listed them all out for clarity
+	// finance.quote has a HUGE variety of info in it that we can pull
+	symbol := quote.Symbol
+	name := quote.ShortName
+	currentPrice := decimal.NewFromFloat(quote.RegularMarketPrice)
+	priceChange := float32(quote.RegularMarketChange)
+	percentChange := float32(quote.RegularMarketChangePercent)
+
+	return &Stock{symbol, name, currentPrice, priceChange, percentChange}, nil
+
+}
+
+// GetHistoricalPrice is hacky because Fingo only has a function that returns quotes in a time span
+// This function takes in the desired date as the end date, then sets the start date as five days prior
+func (piq FingoPack) GetHistoricalPrice(ticker string, date time.Time) (*HistoricalStock, error) {
+
+	quoteAtDate := new(finance.ChartBar)
+
+	// creating a hypothetical start five days before (is this long enough)
+	adjStart := date.AddDate(0, 0, -5)
```

Comment:
🤔 why are we arbitrarily subtracting 5 days now? seems like there's no min requirement for a date range, so why not just +1 and -1 to our date? Better to fail later and understand why than just arbitrarily save unnecessary date ranges